app/models/megaphone/publisher.rb

```diff
@@ -101,7 +101,16 @@ def create_episode!(megaphone_podcast, ep)
       me.create!
     end
 
-    def unpublish_and_delete_episodes!
+    def delete_episodes!
+      megaphone_episodes = []
+      podcast.episodes.with_deleted.where.not(id: private_feed.episodes).each do |ep|
```

Comment:
It looks like this is skipping the HABTM relation for episodes <-> feeds.  It that right? Is there a integration specific reason?

Reply:
Also, in that vein, do we need `private_feed.episodes.with_deleted` here?

Reply:
it is looking at episodes in the podcast, not just the feed, to find episodes removed from the Megaphone feed, which indicates they should be deleted from Megaphone. Those episodes aren't deleted from feeder, but should be deleted from megaphone.

I think there is a case I am missing here though, which is episodes still in the Megaphone feed which have been deleted, I'll add that!